libr/anal/p/anal_mips_gnu.c

```diff
@@ -35,10 +34,8 @@ static int mips_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *b_in, int len
 	op->addr = addr;
 	r_strbuf_init (&op->esil);
 
-	// Reminder: r_mem_copyendian swaps if arg `endian` ==0 ...
-	// When anal->big_endian is "false", as for mipsel architecture, we NEED to swap here for the below analysis to work.
-	r_mem_copyendian ((ut8*)&opcode, b_in, 4, anal->big_endian ? 1 : 0);
-	r_mem_copyendian (b, b_in, 4, anal->big_endian ? 1 : 0);
+	// Be endian aware
+	opcode = r_read_ble32 (b, anal->big_endian);
```

Comment:
I think the test is failing due to libr/anal/p/anal_mips_cs.c the opcodes are not implemented correctly in esil, I think its a signed/unsigned issue with ADDIU instruction..